agent/consul/server.go

```diff
@@ -1141,6 +1175,8 @@ func (s *Server) GetLANCoordinate() (lib.CoordinateSet, error) {
 // ReloadConfig is used to have the Server do an online reload of
 // relevant configuration information
 func (s *Server) ReloadConfig(config *Config) error {
+	s.rpcLimiter.Store(rate.NewLimiter(config.RPCRate, config.RPCMaxBurst))
```

Comment:
Since both client and server are doing the same now, I think we can move this into the agent's `ReloadConfig()`.

Reply:
Hmm, we could although it's actually accessed inside the `RPC` methods which are in the Client/Server delegates. Server doesn't have a back-reference to the agent object currently and i worry adding one kinda breaks the encapsulation of the `consul` package which currently doesn't reason about things "higher up" in the `agent` package.

If it were more that a line of code that was duplicated I'd maybe consider if there is a different way to share the logic but it seems simpler to live with the tiny duplication for now? What do you think?

Reply:
Agreed! I only remembered seeing a very similar line on the client side and assumed that the `rpcLimiter` is available in the agent as well. Since thats not the case, let's leave it like it is.